### PR TITLE
message context menu (WEBAPP-2861)

### PR DESCRIPTION
**Pull request by herrmannplatz**
This adds the new message context menu to the UI. 

The basic idea is that the context menu emits a custom event once its clicked and the data is updated. This way you can listen to the custom event and stop the propagation to e.g. display a different element with the data.

### CODE REVIEW

File: app/script/components/contextMenu.coffee [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see http://www.gnu.org/licenses/.
+#
+
+window.z ?= {}
+z.components ?= {}
+
+z.components.ContextMenuEvent =
+  CONTEXT_MENU: 'customContextMenu'
```
Comment (bennycode) [inline app/script/components/contextMenu.coffee]: Usually we declare custom values like this:

``` coffee
z.components.ContextMenuEvent =
  CONTEXT_MENU: 'z.components.ContextMenuEvent::CONTEXT_MENU'
```

With this we can copy the value `z.components.ContextMenuEvent::CONTEXT_MENU` and search through the code where this value is being used. Makes it easier to find it's reference.

File: app/script/components/contextMenu.coffee [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+# Display context menu for the given entries. Entries should be provided as an Array of object containing
```
Comment (bennycode) [inline app/script/components/contextMenu.coffee]: `Array of objects` (s)

File: app/script/components/contextMenu.coffee [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#
+# entries example:
+# [
+#   {label: 'copy', action: 'action-id'},
+#   {label: 'delete', action: 'action-id'}
+# ]
+#
+# @param [Object] params
+# @option params [String] tag
+# @option params [Object] data additional data that is passed then an items in the context menu is clicked
```
Comment (bennycode) [inline app/script/components/contextMenu.coffee]: `additional data that is passed then an items in the context menu is clicked`
->
`additional data that is passed when items in the context menu are clicked`

File: app/style/components/context-menu.less [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+context-menu {
```
Comment (lipis) [inline app/style/components/context-menu.less:20]: `.context-menu` or I'm missing something?

Comment (herrmannplatz) [inline app/style/components/context-menu.less:20]: nope its the custom element itself

Comment (lipis) [inline app/style/components/context-menu.less:20]: just wanted to verify that I'm missing something :D 🏇

**Pull request merged**